src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Program.cs

```diff
@@ -202,11 +214,28 @@ static CliRootCommand CreateCommandLineParser()
             SessionId: sessionId,
             SharedDependenciesPath: sharedDependenciesPath,
             ExtensionAssemblyPaths: extensionAssemblyPaths,
-            ExtensionLogDirectory: extensionLogDirectory);
+            ExtensionLogDirectory: extensionLogDirectory,
+            PipeName: pipe);
 
         return RunAsync(serverConfiguration, cancellationToken);
     });
 
     return rootCommand;
 }
 
+static string GetPlatformPipeConnectionString(string pipeName)
+{
+    var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+    if (isWindows)
+    {
+        return @"\\.\" + pipeName;
```

Comment:
I'd similarly vote on this side also just do all the string munging on the client side, so this just takes a string and passes it through without further surprises.

Reply:
Sure, I can move to the client!